src/main/resources/templates/registration_page.html

```diff
@@ -22,18 +22,21 @@
 <body>
 <main>
     <h1 class="mb-4 text-center">Links</h1>
-    <form th:action="@{/register}" method="post" enctype="application/x-www-form-urlencoded" autocomplete="on">
+    <form th:action="@{/register}" th:object="${user}" method="post" enctype="application/x-www-form-urlencoded" autocomplete="on">
         <div class="mb-3">
             <label for="username" class="form-label">Username</label>
-            <input type="text" class="form-control" id="username" aria-describedby="emailHelp" name="username">
+            <input type="text" class="form-control" id="username" aria-describedby="emailHelp" name="username" th:field="*{username}">
+            <div class="text-danger" th:if="${#fields.hasErrors('username')}" th:errors="*{username}">Name error</div>
         </div>
         <div class="mb-3">
             <label for="email" class="form-label">Email address</label>
             <input type="email" class="form-control" id="email" aria-describedby="emailHelp" name="email">
+            <div class="text-danger" th:if="${#fields.hasErrors('email')}" th:errors="*{email}">Email error</div>
```

Comment:
Let's use styles provided by Bootstrap - https://getbootstrap.com/docs/5.3/forms/validation/#server-side
Could you implement this?

Reply:
Working on it...

Reply:
Well... I succeeded to achieve the bootstrap style, it looks exactly how it should. But validation logic crashed somehow.
Don't quite understand how, but I'll figure it out.
Could you please review AuthController.registerUser() and registration_page.html.  If they are ok from stylish point of view, I'll go for org.springframework.dao.DataIntegrityViolationException  fixing.

Reply:
Styles are okay